go/caddy/pfpki/models.go

```diff
@@ -0,0 +1,755 @@
+		Information.Error = err.Error()
+		return Information, CaDB.Error
+	}
+
+	if err := pfpki.DB.Create(&Profile{Name: p.Name, Ca: ca, CaID: p.CaID, CaName: ca.Cn, Validity: p.Validity, KeyType: p.KeyType, KeySize: p.KeySize, Digest: p.Digest, KeyUsage: p.KeyUsage, ExtendedKeyUsage: p.ExtendedKeyUsage, P12MailPassword: p.P12MailPassword, P12MailSubject: p.P12MailSubject, P12MailFrom: p.P12MailFrom, P12MailHeader: p.P12MailHeader, P12MailFooter: p.P12MailFooter}).Error; err != nil {
+		Information.Error = err.Error()
+		return Information, errors.New("A database error occured. See log for details.")
+	}
+	pfpki.DB.Select("id, name, ca_id, ca_name, validity, key_type, key_size, digest, key_usage, extended_key_usage, p12_mail_password, p12_mail_subject, p12_mail_from, p12_mail_header, p12_mail_footer").Where("name = ?", p.Name).First(&profiledb)
+	Information.Entries = profiledb
+
+	return Information, nil
+}
+
+func (p Profile) update(pfpki *Handler) (Info, error) {
+	var profiledb []Profile
+	Information := Info{}
+	if err := pfpki.DB.Model(&Profile{}).Updates(&Profile{P12MailPassword: p.P12MailPassword, P12MailSubject: p.P12MailSubject, P12MailFrom: p.P12MailFrom, P12MailHeader: p.P12MailHeader, P12MailFooter: p.P12MailFooter}).Error; err != nil {
+		Information.Error = err.Error()
+		return Information, errors.New("A database error occured. See log for details.")
+	}
+	pfpki.DB.Select("id, name, ca_id, ca_name, validity, key_type, key_size, digest, key_usage, extended_key_usage, p12_mail_password, p12_mail_subject, p12_mail_from, p12_mail_header, p12_mail_footer").Where("name = ?", p.Name).First(&profiledb)
+	Information.Entries = profiledb
+
+	return Information, nil
+}
+
+func (p Profile) getById(pfpki *Handler, params map[string]string) (Info, error) {
+	Information := Info{}
+	var profiledb []Profile
+	if val, ok := params["id"]; ok {
+		allFields := strings.Join(SqlFields(p)[:], ",")
+		pfpki.DB.Select(allFields).Where("`id` = ?", val).First(&profiledb)
+	}
+	Information.Entries = profiledb
+
+	return Information, nil
+}
+
+func (p Profile) paginated(pfpki *Handler, vars Vars) (Info, error) {
+	Information := Info{}
+	var count int
+	pfpki.DB.Model(&Profile{}).Count(&count)
+	Information.TotalCount = count
+	Information.PrevCursor = vars.Cursor
+	Information.NextCursor = vars.Cursor + vars.Limit
+	if vars.Cursor < count {
+		sql, err := vars.Sql(p)
+		if err != nil {
+			Information.Error = err.Error()
+			return Information, errors.New("A database error occured. See log for details.")
+		}
+		var profiledb []Profile
+		pfpki.DB.Select(sql.Select).Order(sql.Order).Offset(sql.Offset).Limit(sql.Limit).Find(&profiledb)
+		Information.Entries = profiledb
+	}
+
+	return Information, nil
+}
+
+func (p Profile) search(pfpki *Handler, vars Vars) (Info, error) {
+	Information := Info{}
+	sql, err := vars.Sql(p)
+	if err != nil {
+		Information.Error = err.Error()
+		return Information, errors.New("A database error occured. See log for details.")
```

Comment:
It doesn't look like the error is actually logged unless I'm missing some magic

Reply:
gorm logs the sql error in the logs, it's just for the frontend.
Btw i need to work on the loglevel.